patches/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java.patch

```diff
@@ -0,0 +1,11 @@
+--- ../src-base/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java
++++ ../src-work/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java
+@@ -94,6 +94,8 @@
+                             world.func_180501_a(blockpos, Blocks.field_185773_cZ.func_176223_P(), 3);
+                             flag = true;
+                         }
++                        else if (itemstack.func_77973_b() instanceof net.minecraftforge.common.IPlantable && ((net.minecraftforge.common.IPlantable)itemstack.func_77973_b()).getPlantType(world, blockpos)==net.minecraftforge.common.EnumPlantType.Crop) {
++                        	world.func_180501_a(blockpos, ((net.minecraftforge.common.IPlantable)itemstack.func_77973_b()).getPlant(world, blockpos), 3); flag = true; }
```

Comment:
This should probably be 3 lines, putting the closing brace at the end of a statement makes this section hard to understand.